spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -226,19 +230,118 @@ local | 1
     thrown(MissingPropertyException)
   }
 
-  def "cells cannot reference other cells"() {
+  def 'cells can reference previous cells'() {
+    expect:
+    [a, b, c] == [0, 1, 2]
+
+    where:
+    a | b     | c
+    0 | a + 1 | b + 1
+  }
+
+  def 'cell references are pointing to the current row'() {
+    expect:
+    b == 1 + a * 2
+    c == 3 * b
+
+    where:
+    a | b         | c
+    0 | 1 + a * 2 | 3 * b
+    1 | 1 + a * 2 | 3 * b
+    2 | 1 + a * 2 | 3 * b
+  }
+
+  def "cell references are evaluated correctly in the method's name"() {
     when:
-    runner.runFeatureBody """
-expect:
-true
+    runner.runSpecBody '''
+      @Unroll def 'a = #a, b = #b'() {
+        expect:
+        true
+        
+        where:
+        a | b
+        0 | a + 1
+      }
+    '''
 
-where:
-a | b
-1 | a
-    """
+    then:
+    1 * listener.testStarted { it.methodName == "a = 0, b = 1" }
+  }
+
+  def "cells can't reference next cells"() {
+    when:
+    runner.runFeatureBody '''
+      expect:
+      false
+      
+      where:
+      a | b
+      b | 1
+    '''
 
     then:
-    thrown(MissingPropertyException)
+    thrown Exception
+  }
+
+  def "cells can't reference themselves"() {
+    when:
+    runner.runFeatureBody '''
+      expect:
+      false
+      
+      where:
+      a | b
+      1 | b + 1
+    '''
+
+    then:
+    thrown Exception
+  }
+
+  def 'cell references are working with simple parameterization also'() {
+    expect:
+    c == 2
+
+    where:
+    a << [1, 2]
+    b << [3, 4]
+    c << [b - a, b - a]
+  }
+
+  def 'data tables can be referenced from following variables'() {
+    expect:
+    c == 3
+
+    where:
+    a | b
+    1 | 2
+    1 | a + 1
+
+    c = b + 1
+  }
+
+  @Ignore
+  def 'data table elements can reference each other'() {
+    expect:
+    runner.runFeatureBody '''
+      expect:
+      g == 12
+  
+      where:
+      a = 1
+      b = a + 1
+      
+      c << [b + 1]
```

Comment:
Can this be fixed? Otherwise the user will have to know which combinations work and which don't.
Also it might be a good idea to have more simple tests for each of the combinations used in this test, some exist already but there are still missing combinations. Especially the minimal failing `where` clause.